app/database/queries/breeze_buddy/user_memory.py

```
@@ -0,0 +1,296 @@
+) -> Tuple[str, List[Any]]:
+    """Supersede one record while preserving the foundation's public query API."""
+    text = f"""
+        UPDATE "{USER_MEMORY_TABLE}"
+        SET superseded_at = now(), updated_at = now()
+        WHERE reseller_id = $1
+          AND merchant_id = $2
+          AND customer_key = $3
+          AND id = $4
+          AND superseded_at IS NULL
+        RETURNING {_FACT_COLUMNS};
+    """
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)